transport/pool.go

```diff
@@ -118,6 +129,13 @@ func (r *PoolRefiller) initConnPool() error {
 	}
 	ss := s.ScyllaSupported()
 
+	if s, ok := s.Options["SCYLLA_SHARD_AWARE_PORT"]; ok && !strings.HasSuffix(r.addr, ":"+s[0]) {
```

Comment:
I would consider renaming variable s since it has already been used (with different meaning) inside this scope.

Moreover, we have const defined for string "SCYLLA_SHARD_AWARE_PORT" inside response/supported.go.
I would consider using it.